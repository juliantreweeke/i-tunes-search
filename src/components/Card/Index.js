import PropTypes from "prop-types";
import React from 'react';
import styles from "./card.module.css";
import Heading from "../Heading/Index";

const Card = ({ data, onClick }) => {

  return (
    <li onClick={onClick} className={styles.card}>
      <a href={data.url}>
        <div className={styles.imageContainer}>
          <img
            src={data.image}
            alt={data.collectionName}
          />
        </div>
        <div className={styles.text}> 
          <Heading size="medium" className={styles.heading} headingLevel="h2">
            {data.heading}
          </Heading>
          <p>{data.text}</p>
        </div>
        <div className={styles.detailContainer}>
          <small>{data.detail}</small>
        </div>  
      </a>
    </li>
  )};

Card.propTypes = {
  data: PropTypes.object.isRequired,
  onClick: PropTypes.func,
};

export default Card;
