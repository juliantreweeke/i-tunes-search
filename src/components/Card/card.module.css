.card {
    border: 1px solid;
    border-radius: var(--radius);
    min-height: 300px;
}

.card a {
  display: block;
  height: 100%;
  display: flex;
  flex-direction: column;
  flex-grow: 1;
}

.card .imageContainer {
    height: 6.5rem;
    clip-path: polygon(0 0, 100% 0, 100% 100%, 0 calc(100% - 1rem)); 
}

.card .imageContainer img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    border-radius: var(--radius);
}

.card .text, .card .dateContainer {
  color: var(--color-text-dark);
  padding: var(--component-padding);
  max-width: 60ch;
}

.card .dateContainer {
  margin-top: auto;
}

.card:hover {
    box-shadow: 0 0 0 0.25rem;
}



