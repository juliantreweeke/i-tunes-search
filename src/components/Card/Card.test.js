import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Card from "./Index";
import { axe, toHaveNoViolations } from "jest-axe";

const mockData = {
  url: "test.com",
  heading: "heading",
  image: "testimage.com",
  text: "some text for testing",
  detail: "some detail text for testing",
};

describe("Card", () => {
  it("should render", () => {
    const { container } = render(<Card data={mockData} />);
    expect(container).toBeTruthy();
  });

  it("matches snapshot with data", () => {
    const { container } = render(<Card data={mockData} />);
    expect(container).toMatchSnapshot();
  });

  it("when component clicked, fires onClick prop", () => {
    const onClickFunction = jest.fn();
    render(<Card data={mockData} onClick={onClickFunction} />);
    fireEvent.click(screen.getByRole("listitem"));
    expect(onClickFunction).toBeCalled();
  });

  it("When wrapped in a <ul> tag, component should have no accessibility violations", async () => {
    expect.extend(toHaveNoViolations);
    const { container } = render(
      <ul>
        <Card data={mockData} />
      </ul>
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
