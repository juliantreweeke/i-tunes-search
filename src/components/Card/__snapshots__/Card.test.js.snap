// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot with data 1`] = `
<div>
  <li
    class="card"
  >
    <a
      href="test.com"
    >
      <div
        class="imageContainer"
      >
        <img
          alt="heading"
          src="testimage.com"
        />
      </div>
      <div
        class="text"
      >
        <h2
          class="heading"
          data-size="medium"
        >
          heading
        </h2>
        <p>
          some text for testing
        </p>
      </div>
      <div
        class="detailContainer"
      >
        <small>
          some detail text for testing
        </small>
      </div>
    </a>
  </li>
</div>
`;
